sql/core/src/main/java/org/apache/spark/sql/sources/v2/SupportsBatchWrite.java

```diff
@@ -25,14 +25,14 @@
 import org.apache.spark.sql.types.StructType;
 
 /**
- * A mix-in interface for {@link DataSourceV2}. Data sources can implement this interface to
+ * A mix-in interface for {@link Table}. Data sources can implement this interface to
  * provide data writing ability for batch processing.
  *
  * This interface is used to create {@link BatchWriteSupport} instances when end users run
  * {@code Dataset.write.format(...).option(...).save()}.
  */
 @Evolving
-public interface BatchWriteSupportProvider extends DataSourceV2 {
+public interface SupportsBatchWrite extends Table {
```

Comment:
It is quite confusing to have `BatchWriteSupport` and  `SupportsBatchWrite` to me.

Reply:
That's why I left https://github.com/apache/spark/pull/23208#discussion_r238524973 .

namings are welcome!

Reply:
`Table` exposes `newScanBuilder` without an interface. Why should the write side be different? Doesn't Spark support sources that are read-only and write-only?

I think that both reads and writes should use interfaces to mix support into `Table` or both should be exposed by `Table` and throw `UnsupportedOperationException` by default, not a mix of the two options.

Reply:
I do think read-only or write-only is a necessary feature, according to what I've seen in the dev list. Maybe we should move `newScanBuilder` from `Table` to the mixin traits.

Reply:
I'm fine either way, as long as we are consistent between the read and write sides.